torch/optim/optimizer.py

```diff
@@ -32,6 +35,24 @@ def _use_grad(self, *args, **kwargs):
         return ret
     return _use_grad
 
+def _get_value(x):
+    # item is significantly faster than a cpu tensor in eager mode
+    if not torch.jit.is_scripting() and is_compiling():
```

Comment:
Why scripting is involved here?

Reply:
Yeah the tests last night were failing. TorchScript expects each variable to only have a single type and each function to only have a single return type, so I gave TorchScript this "type hint" essentially.